src/filters/examples/filter-panel.js

```diff
@@ -0,0 +1,363 @@
+                id: 'val2',
+                title: 'Value 2',
+                value: 'Value 2',
+                selected: false
+              },
+              {
+                id: 'val3',
+                title: 'Value 3',
+                value: 'Value 3',
+                selected: false
+              }
+            ]
+          },
+          {
+            id: 'category2',
+            title:  'Category Two',
+            filterType: 'checkbox',
+            values: [
+              {
+                id: 'val1',
+                title: 'Value 1',
+                value: 'Value 1',
+                selected: false
+              },
+              {
+                id: 'val2',
+                title: 'Value 2',
+                value: 'Value 2',
+                selected: false
+              },
+              {
+                id: 'val3',
+                title: 'Value 3',
+                value: 'Value 3',
+                selected: false
+              }
+            ]
+          }
+        ];
+
+        $scope.filtersText = '';
+
+        $scope.allItems = [
+          {
+            id: "1",
+            keyword: "Foo",
+            categoryOne: "Value 1",
+            categoryTwo: "Value 1"
+          },
+          {
+            id: "2",
+            keyword: "Bar",
+            categoryOne: "Value 2",
+            categoryTwo: "Value 1"
+          },
+          {
+            id: "3",
+            keyword: "Foo",
+            categoryOne: "Value 3",
+            categoryTwo: "Value 1"
+          },
+          {
+            id: "4",
+            keyword: "Bar",
+            categoryOne: "Value 1",
+            categoryTwo: "Value 2"
+          },
+          {
+            id: "5",
+            keyword: "Foo",
+            categoryOne: "Value 2",
+            categoryTwo: "Value 2"
+          },
+          {
+            id: "6",
+            keyword: "Bar",
+            categoryOne: "Value 3",
+            categoryTwo: "Value 2"
+          },
+          {
+            id: "7",
+            keyword: "Foo",
+            categoryOne: "Value 1",
+            categoryTwo: "Value 3"
+          },
+          {
+            id: "8",
+            keyword: "Bar",
+            categoryOne: "Value 2",
+            categoryTwo: "Value 3"
+          },
+          {
+            id: "9",
+            keyword: "Foo",
+            categoryOne: "Value 3",
+            categoryTwo: "Value 3"
+          }
+        ];
+        $scope.items = $scope.allItems;
+
+        // called when filter cleared by hitting 'x' in filter results tag, or 'Clear All Filters' link
+        var onFilterChange = function (appliedFilters, changedFilterId, changedFilterValue) {
+          if (angular.isDefined(changedFilterId) && angular.isDefined(changedFilterValue)) {
+            updateFilterPanelModel(changedFilterId, changedFilterValue);
+          } else {
+            // the 'Clear All Filters' link was clicked
+            resetFilterPanelModel();
```

Comment:
It they click on the 'x' for all the values of all filter, then the appliedFilters list would be empty, but they didn't click on the clear all.  I could program it that way, but it's less efficient in the scenario I just mentioned because clicking on the last 'x' would trigger resetFilterPanelModel() which goes through the entire model (as opposed to just clearing the last filter value whose 'x' was clicked)